### ISSUE TITLE
Add support for AWS App Mesh Virtual Gateway

### ISSUE THREAD

**Issue by completenovice** [terraform-providers/terraform-provider-aws#14169]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Allow the creation of an AWS App Mesh Virtual Gateway.

### New or Affected Resource(s)

* aws_appmesh_virtual_gateway
* aws_appmesh_gateway_route

### Potential Terraform Configuration

```hcl
resource "aws_appmesh_virtual_gateway" "vgateway" {
  name      = "example-gateway"
  mesh_name = "mesh"

  spec {
    backendDefaults {
      clientPolicy {
        tls {
          enforce = false
        }
      }
    }

    listeners = [
      {
        port_mapping {
          port     = 80,
          protocol = "http",
        }
      }
    ]
  }
}

resource "aws_appmesh_gateway_route" {
  gateway_route_name   = "http_route"
  virtual_gateway_name = aws_appmesh_virtual_gateway.vgateway.name
  mesh_name            = "mesh"

  spec {
    http_route {
      action {
        target {
          virtual_service {
            virtual_service_name = "my-virtual-service"
          }
        }
      }
      match {
        prefix = "/"
      }
    }
  }
}
```

### References

* https://aws.amazon.com/blogs/containers/introducing-ingress-support-in-aws-app-mesh/
* https://github.com/aws/aws-app-mesh-roadmap/issues/37#issuecomment-657918106

**Comment by ewbankkit** [comment]
Requires [AWS SDK v1.33.5](https://github.com/aws/aws-sdk-go/releases/tag/v1.33.5):

* #14000

**Comment by rmaskara** [comment]
when are we planning take this up?

**Comment by ewbankkit** [comment]
@rmaskara We are working through the backlog of AppMesh PRs, many of which were released last week and with #14361 at the last outstanding one. I will start work on this issue shortly.

**Comment by rmaskara** [comment]
Thanks @ewbankkit . We are using the stuff that was released last week already. So, that was very helpful! Would you say that it would take about a couple of months to get it through development and PR approvals?
